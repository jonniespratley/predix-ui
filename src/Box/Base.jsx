import React from 'react';
import PropTypes from 'prop-types';
import use from 'reuse';

const Base = ({ use: T = 'div', ...props }) => <T {...props} />;

Base.propTypes = {
  use: PropTypes.string
};

Base.defaultProps = {
  use: 'div'
};
export default use(Base);
