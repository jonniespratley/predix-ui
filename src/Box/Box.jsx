import use from 'reuse';

import Base from './Base';

const Box = use(Base);

export default Box;
