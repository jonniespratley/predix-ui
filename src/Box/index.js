export { default as Box } from './Box';
export { default as Base } from './Base';
