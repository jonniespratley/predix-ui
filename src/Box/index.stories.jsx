import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import { withReadme } from 'storybook-readme';

import README from './README.md';
import { Box, Base } from '.';

storiesOf('Components / Box', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .addWithJSX('with box', () => (
    <Box use={text('use', 'div')}>
      {text('content', 'This is the main context area of the box.')}
    </Box>
  ))
  .addWithJSX('with base', () => (
    <Base use={text('use', 'h2')}>
      {text('content', 'This is the main context area of the box.')}
    </Base>
  ));

