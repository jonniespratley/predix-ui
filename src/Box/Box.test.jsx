import React from 'react';
import renderer from 'react-test-renderer';

import Box from './Box';

describe('Box', () => {
  test('should render', () => {
    const tree = renderer.create(<Box use="span">Content</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render as <h2>Heading</h2>', () => {
    const tree = renderer.create(<Box use="h2">Content</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
