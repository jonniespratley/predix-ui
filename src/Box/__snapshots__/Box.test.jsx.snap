// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render 1`] = `
<span>
  Content
</span>
`;

exports[`Box should render as <h2>Heading</h2> 1`] = `
<h2>
  Content
</h2>
`;
